### ISSUE TITLE
Suggestion: cancelAll if Android id badge count is zero

### ISSUE THREAD

**Issue by chriscant** [phonegap/phonegap-plugin-push#2520]
### Expected Behaviour

We have a use case which is probably fairly common where notification(s) need to removed when an action has been completed elsewhere, ie a work item has been done on the web so the app notification had best be removed.  A bit like: you've read a Facebook chat message in a browser so it can be cleared from the app.

I expect this to happen even when the app isn't running (even in the background).

My app code makes it work as intended when running in foreground or background ie it can see when count==0 and call clearAllNotifications(). However it doesn't work as expected when my app isn't running at all.

### Actual Behaviour

It works as expected on iOS but doesn't on Android.

#### Steps to Reproduce 

Send a notification with count=1 and a title say.
Sending a further notification with count=0 removes the notification in iOS but not Android.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

Android 8.1 and 6.

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)

Seen on Nexus 5X and Alcatel Pixi.

#### Cordova CLI version and cordova platform version

Cordova 8.0.0
Cordova android 6.3.0

#### Plugin version

I've reproduced the problem and found workarounds in the plugin 1.11.1 (GCM) and 2.1.3 (Firebase).

#### Sample Push Data Payload

var message = {
    data: {
      title: "All work done",
      message: "date-time"
      count: count.toString(),
      image: "www/img/something.png",
      "content-available": '1',
      notId: Math.floor(new Date().getTime() / 1000).toString()
    },
    apns: {
      payload: {
        aps: {
          badge: count,
        },
      },
    },
    token: token
  };

### Suggested fix

I've tested additions in the function showNotificationIfPossible() in both 1.11.1 GCMIntentService.java and 2.1.3 FCMService.java

```
if (badgeCount == 0) {
  NotificationManager mNotificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
  mNotificationManager.cancelAll();
  return;
}

```
Note that badgeCount will be -1 if a count is not given

It may be that more generally developers might only want the notification cleared if the text is empty as well.

**Comment by macdonst** [comment]
@chriscant seems like a great and simple addition for a PR. Could you send one so we can include it in 3.0.0?

**Comment by chriscant** [comment]
@macdonst will do - probably tomorrow

**Comment by chriscant** [comment]
My suggested code as it stands stops the background notification event handler being called.
Removing the return probably fixes it. I'll test that tomorrow.

**Comment by macdonst** [comment]
To be included in 3.0.0
